frontend/src/features/admin-form/settings/components/AuthSettingsSection/SecretKeyDownloadWhitelistFileModal.tsx

```diff
@@ -0,0 +1,282 @@
+import { useCallback, useRef } from 'react'
+import { useForm } from 'react-hook-form'
+import { BiRightArrowAlt, BiUpload } from 'react-icons/bi'
+import { useQueryClient } from 'react-query'
+import {
+  Container,
+  FormControl,
+  FormLabel,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  Stack,
+  useBreakpointValue,
+  UseDisclosureReturn,
+} from '@chakra-ui/react'
+import Papa from 'papaparse'
+
+import {
+  decryptStringMessage,
+  EncryptedStringsMessageContent,
+} from '~shared/utils/crypto'
+
+import { useToast } from '~hooks/useToast'
+import { isKeypairValid, SECRET_KEY_REGEX } from '~utils/secretKeyValidation'
+import Button from '~components/Button'
+import { downloadFile } from '~components/Field/Attachment/utils/downloadFile'
+import { FormErrorMessage } from '~components/FormControl/FormErrorMessage/FormErrorMessage'
+import IconButton from '~components/IconButton'
+import Input from '~components/Input'
+
+import { fetchAdminFormEncryptedWhitelistedSubmitterIds } from '../../queries'
+import { FormActivationSvg } from '../FormActivationSvg'
+
+export interface SecretKeyDownloadWhitelistFileModalProps
+  extends Pick<UseDisclosureReturn, 'onClose' | 'isOpen'> {
+  publicKey: string
+  formId: string
+  downloadFileName: string
+}
+
+const SECRET_KEY_NAME = 'secretKey'
+
+interface SecretKeyFormInputs {
+  [SECRET_KEY_NAME]: string
+}
+
+const useSecretKeyWhitelistFileModal = ({
```

Comment:
/nit can consider moving this to it's own file `useSecretKeyWhitelistFileModal.tsx` in the same folder